File: leetcode/9.js
/**
 * Title: 9. Palindrome Number
 * URL: https://leetcode.com/problems/palindrome-number/
 * Answered by @referanslar
 */

/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  let arr = String(x).split("");
  let reversedArr = arr.reverse();
  let reversed = reversedArr.join("");

  return x == reversed;
};

File: leetcode/9.c
/**
 * Title: 9. Palindrome Number
 * URL: https://leetcode.com/problems/palindrome-number/
 * Answered by @referanslar
 */

bool isPalindrome(int x)
{
    int original = x;
    long long reversed = 0;

    while (x > 0)
    {
        int digit = x % 10;
        reversed = reversed * 10 + digit;
        x /= 10;
    }

    return original == reversed;
}
